calico_upgrade/pkg/commands/validate.go

```diff
@@ -52,8 +51,65 @@ Description:
 		fmt.Printf("Invalid option: 'calicoctl %s'. Use flag '--help' to read about a specific subcommand.\n", strings.Join(args, " "))
 		os.Exit(1)
 	}
-	if len(parsedArgs) == 0 {
-		return
+	cfv2 := parsedArgs["--calicov2-config"].(string)
+	cfv3 := parsedArgs["--calicov3-config"].(string)
+
+	_, clientV2, err := upgradeclients.LoadClients(cfv3, cfv2)
+	if err != nil {
+		fmt.Printf("Failed to create Calico API client: %s\n", err)
+		os.Exit(1)
+	}
+
+	cData, err := migrate.QueryAndConvertResources(clientV2)
+
+	// Any resource names we plan to change should be reported to the user.
+	if len(cData.NameConversions) > 0 {
+		fmt.Println("The following resource names will be changed:")
+		for _, change := range cData.NameConversions {
+			fmt.Printf(" -  %s: %s -> %s\n", change.Kind, change.Original, change.New)
+		}
+	}
+
+	// After we've converted the names we might end up with clashing names that the user will need to update.
+	if len(cData.NameClashes) > 0 {
+		fmt.Println("The following names clashed after conversion was applied:")
+		for _, nError := range cData.NameClashes {
+			fmt.Println(nError)
```

Comment:
might be nice to prepend a \<tab\> for each clashing name, or better yet, use the same format you use on line 69: `(" -  %s")`.

Reply:
The `String()` method for these has that sort of formatting in it already see https://github.com/projectcalico/calico/blob/calico-upgrade/calico_upgrade/pkg/migrate/migrate.go#L99-L126

Reply:
ah, cool. thanks for the pointer.